samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -69,25 +69,28 @@
 
   private final StreamGraphBuilder graphBuilder;
 
+  private final ApplicationRunner runner;
+
   private ContextManager contextManager;
 
-  public StreamOperatorTask(StreamGraphBuilder graphBuilder) {
+  public StreamOperatorTask(StreamGraphBuilder graphBuilder, ApplicationRunner runner) {
     this.graphBuilder = graphBuilder;
+    this.runner = runner;
```

Comment:
It's slightly strange to have `ApplicationRunner` in the constructor. All you need is to be able to get a `StreamSpec` for a given `streamId`.  We probably need a separate interface for that. 

Long term we want to be able to serialize this, and make the resolution more obvious.